how to can exclude grpc-inprocess

**The context**

What do you wish to achieve?

I want to exclude grpc-inprocess in client.
```
    <dependency>
      <groupId>net.devh</groupId>
      <artifactId>grpc-client-spring-boot-starter</artifactId>
      <version>3.1.0.RELEASE</version>
      <exclusions>
        <exclusion>
          <artifactId>grpc-inprocess</artifactId>
          <groupId>io.grpc</groupId>
        </exclusion>
      </exclusions>
    </dependency>
```

**The bug**

What's the problem? What's not working? What do you expect to happen.

when i run my App, it will fail with Exception.

Because the  `InProcessChannelFactory` is hard code into  GrpcClientAutoConfiguration.

**Stacktrace and logs**

Is there a stacktrace or a hint in the logs? (very important)
Screenshots work as well, but don't screenshot your logs.
```
Caused by: org.springframework.beans.BeanInstantiationException: Failed to instantiate [net.devh.boot.grpc.client.channelfactory.GrpcChannelFactory]: Factory method 'shadedNettyGrpcChannelFactory' threw exception with message: io/grpc/inprocess/InProcessChannelBuilder
	at org.springframework.beans.factory.support.SimpleInstantiationStrategy.instantiate(SimpleInstantiationStrategy.java:177)
	at org.springframework.beans.factory.support.ConstructorResolver.instantiate(ConstructorResolver.java:644)
	... 36 common frames omitted
Caused by: java.lang.NoClassDefFoundError: io/grpc/inprocess/InProcessChannelBuilder
	at net.devh.boot.grpc.client.autoconfigure.GrpcClientAutoConfiguration.shadedNettyGrpcChannelFactory(GrpcClientAutoConfiguration.java:160)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.springframework.beans.factory.support.SimpleInstantiationStrategy.instantiate(SimpleInstantiationStrategy.java:140)
	... 37 common frames omitted
Caused by: java.lang.ClassNotFoundException: io.grpc.inprocess.InProcessChannelBuilder
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:641)
	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:188)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:525)
	... 43 common frames omitted

```

**Steps to Reproduce**

Steps to reproduce the behavior:
1. create spring boot App with stub client
2. exclude grpc-inprocess
3. run the App


**The application's environment**

Which versions do you use?
* Spring (boot):
* grpc-java:
* grpc-spring-boot-starter:
* java: version + architecture (64bit?)
* Other relevant libraries...

**Additional context**

* Did it ever work before?
* Do you have a demo?

--- a/grpc-client-spring-boot-starter/src/main/java/net/devh/boot/grpc/client/autoconfigure/GrpcClientAutoConfiguration.java
+++ b/grpc-client-spring-boot-starter/src/main/java/net/devh/boot/grpc/client/autoconfigure/GrpcClientAutoConfiguration.java
@@ -143,13 +143,13 @@ public class GrpcClientAutoConfiguration {
         return Collections.emptyList();
     }
 
-    // First try the shaded netty channel factory
+    // First try the shaded netty channel factory and in process factory
     @ConditionalOnMissingBean(GrpcChannelFactory.class)
     @ConditionalOnClass(name = {"io.grpc.netty.shaded.io.netty.channel.Channel",
-            "io.grpc.netty.shaded.io.grpc.netty.NettyChannelBuilder"})
+            "io.grpc.netty.shaded.io.grpc.netty.NettyChannelBuilder", "io.grpc.inprocess.InProcessChannelBuilder"})
     @Bean
     @Lazy
-    GrpcChannelFactory shadedNettyGrpcChannelFactory(
+    GrpcChannelFactory inProcessOrShadedNettyGrpcChannelFactory(
             final GrpcChannelsProperties properties,
             final GlobalClientInterceptorRegistry globalClientInterceptorRegistry,
             final List<GrpcChannelConfigurer> channelConfigurers) {
@@ -162,12 +162,13 @@ public class GrpcClientAutoConfiguration {
         return new InProcessOrAlternativeChannelFactory(properties, inProcessChannelFactory, channelFactory);
     }
 
-    // Then try the normal netty channel factory
+    // Then try the normal netty channel factory and in process factory
     @ConditionalOnMissingBean(GrpcChannelFactory.class)
-    @ConditionalOnClass(name = {"io.netty.channel.Channel", "io.grpc.netty.NettyChannelBuilder"})
+    @ConditionalOnClass(name = {"io.netty.channel.Channel", "io.grpc.netty.NettyChannelBuilder",
+            "io.grpc.inprocess.InProcessChannelBuilder"})
     @Bean
     @Lazy
-    GrpcChannelFactory nettyGrpcChannelFactory(
+    GrpcChannelFactory inProcessOrNettyGrpcChannelFactory(
             final GrpcChannelsProperties properties,
             final GlobalClientInterceptorRegistry globalClientInterceptorRegistry,
             final List<GrpcChannelConfigurer> channelConfigurers) {
@@ -180,8 +181,38 @@ public class GrpcClientAutoConfiguration {
         return new InProcessOrAlternativeChannelFactory(properties, inProcessChannelFactory, channelFactory);
     }
 
+    // Then try the shaded netty channel factory
+    @ConditionalOnMissingBean(GrpcChannelFactory.class)
+    @ConditionalOnClass(name = {"io.grpc.netty.shaded.io.netty.channel.Channel",
+            "io.grpc.netty.shaded.io.grpc.netty.NettyChannelBuilder"})
+    @Bean
+    @Lazy
+    GrpcChannelFactory shadedNettyGrpcChannelFactory(
+            final GrpcChannelsProperties properties,
+            final GlobalClientInterceptorRegistry globalClientInterceptorRegistry,
+            final List<GrpcChannelConfigurer> channelConfigurers) {
+
+        log.info("Detected grpc-netty-shaded: Creating ShadedNettyChannelFactory");
+        return new ShadedNettyChannelFactory(properties, globalClientInterceptorRegistry, channelConfigurers);
+    }
+
+    // Then try the normal netty channel factory
+    @ConditionalOnMissingBean(GrpcChannelFactory.class)
+    @ConditionalOnClass(name = {"io.netty.channel.Channel", "io.grpc.netty.NettyChannelBuilder"})
+    @Bean
+    @Lazy
+    GrpcChannelFactory nettyGrpcChannelFactory(
+            final GrpcChannelsProperties properties,
+            final GlobalClientInterceptorRegistry globalClientInterceptorRegistry,
+            final List<GrpcChannelConfigurer> channelConfigurers) {
+
+        log.info("Detected grpc-netty: Creating NettyChannelFactory");
+        return new NettyChannelFactory(properties, globalClientInterceptorRegistry, channelConfigurers);
+    }
+
     // Finally try the in process channel factory
     @ConditionalOnMissingBean(GrpcChannelFactory.class)
+    @ConditionalOnClass(name = {"io.grpc.inprocess.InProcessChannelBuilder"})
     @Bean
     @Lazy
     GrpcChannelFactory inProcessGrpcChannelFactory(

